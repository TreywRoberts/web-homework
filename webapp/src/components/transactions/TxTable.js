import React, {Fragment, useState} from 'react';
import { arrayOf, string, bool, number, shape } from 'prop-types';
import { css } from '@emotion/core';

import EditRow from '../rows/EditRow';
import ReadRow from '../rows/ReadRow';

const styles = css`

  form {
    display: flex;
    flex-direction: column;
    // justify-content:center;
    // align-items: center;
    gap: 10px;
    padding: 1rem;
  }
  
  table {
    border-collapse: collapse;
    width: 100%;
  }
  
  th,
  td {
    border: 1px solid #ffffff;
    text-align: center;
    padding: 8px;
    font-size: 2rem;
  }
  
  th {
    background-color: #f6f4e9;
  }
  
  td {
    background-color: #f2f5fb;
  }
  
  form {
    display: flex;
    gap: 5px;
  }
  
  form td:last-child {
    display: flex;
    justify-content: space-evenly;
  }
  
  form * {
    font-size: 28px;
  }
 
`


export function TxTable ({data }) {
  
  const [editFormData, setEditFormData] =useState({
    id:"",
    user_id: "",
    description:"",
    merchant_id:"",
    debit: false,
    credit: false,
    amount: 0
  })
  const [editContactId, setEditContactId] = useState(null)

  const [amount, setAmount] = useState(0)

   const handleRomanConvertClick = (e, num)=>{
      e.preventDefault()
      if(Number.isInteger(amount)){
        setAmount(RomanConverter(num))
      }else{
        setAmount(amount)
      }
    }

    console.log(amount)


  const handleEditFormChange = (e) =>{
    e.preventDefault();

    const fieldName = e.target.getAttribute("id")
    const fieldValue = e.target.value

    const newFormData = {...editFormData};
    newFormData[fieldName] = fieldValue

    setEditFormData(newFormData)
  }

  const handleEditClick =(event, tx) =>{
    event.preventDefault();
    setEditContactId(tx.id)

    const formValues ={
    id: tx.id,
    user_id: tx.user_id,
    description:tx.description,
    merchant_id:tx.merchant_id,
    debit: tx.debit,
    credit: tx.credit,
    amount: tx.amount
    }

    setEditFormData(formValues)

  }

  const handleCancel = () =>{
    setEditContactId(null)
  }
  




  return (
    <div className='app-container'>
    <form>
    <table css={styles}>
      <tbody>
        <tr className='header'>
          {/* <td >ID</td> */}
          <th >User ID</th>
          <th >Description</th>
          <th >Merchant ID</th>
          <th >Debit</th>
          <th>Credit</th>
          <th >Amount</th>
          <th>Actions</th>
        </tr>
        {
          data.map(tx => {
            return(
              <Fragment>
                {editContactId === tx.id ? (
                <EditRow editFormData={editFormData} handleEditFormChange={handleEditFormChange} handleCancel={handleCancel} /> ): (
                <ReadRow tx={tx} handleEditClick={handleEditClick} />)}
              </Fragment>

            )
          })
        }
      </tbody>
    </table>
    </form>
    </div>
  )
}

TxTable.propTypes = {
  data: arrayOf(shape({
    id: string,
    user_id: string,
    description: string,
    merchant_id: string,
    debit: bool,
    credit: bool,
    amount: number
  }))
}
