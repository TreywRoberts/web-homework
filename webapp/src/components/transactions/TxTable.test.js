import { render, screen, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import userEvent from '@testing-library/user-event';
import 'jest-dom/extend-expect';
import { EDIT_TRANSACTION, DELETE_TRANSACTION } from '../../gql/mutation';
import { TxTable } from './TxTable';
import { transactions} from '../../../mocks/transactions-data';

      const mockData = 
        {
        request: {
            query: EDIT_TRANSACTION,
            variables:{
              id: '5d5c1f747e01cd704f18f864',
              user_id: 'Trey',
              description: 'refund',
              merchant_id: 'Target',
              debit: false,
              credit: true,
              amount: 250,
            },      
          },
          newData: jest.fn(()=>({
          data:
            {
              id: '5d5c1f747e01cd704f18f864',
            },
          })),
        }
  
  
      const deleteMock ={
        request:{
          query: DELETE_TRANSACTION,
          variables: {
            id: '5d5c1f747e01cd704f18f863',
          }
        },
        newData: jest.fn(()=>({
          data:{
            id: '5d5c1f747e01cd704f18f863'
          }
        })),
      }

    

describe('Transactions Table', () => {

  it('Should edit a transaction', async  () => {
    const {container} = render(
    <MockedProvider addTypename={false} mocks={[mockData]}>
      <TxTable data={transactions} />
    </MockedProvider>
    )

    const button = screen.getByTestId('transaction-5d5c1f747e01cd704f18f864-editButton');
    userEvent.click(button);

    const userInput = screen.getByDisplayValue(/employee3/i);
    userEvent.clear(userInput);
    userEvent.type(userInput, "Trey");

    const merchantInput = screen.getByDisplayValue(/walmart/i);
    userEvent.clear(merchantInput);
    userEvent.type(merchantInput, "Target")

    const newUser = await screen.findByDisplayValue(/trey/i);
    expect(newUser).toBeInTheDocument();

    const newMerchant = await screen.findByDisplayValue(/target/i);
    expect(newMerchant).toBeInTheDocument();
  
    const saveButton = screen.getByText(/save/i);
    userEvent.click(saveButton);

    const editTransactionMutation = mockData.newData;
    await waitFor(() => expect(editTransactionMutation).toHaveBeenCalled());
  })

  it('Should change digit to roman numeral', ()=>{
    render(
      <MockedProvider addTypename={false} mocks={[mockData]}>
        <TxTable data={transactions} />
      </MockedProvider>
      )

      const numeralButton = screen.getByTestId('transaction-5d5c1f747e01cd704f18f863-numeralButton');

      const digit = screen.getByTestId('transaction-5d5c1f747e01cd704f18f863-amount');
      expect(digit).toHaveTextContent(150);

      userEvent.click(numeralButton);
      expect(digit).toHaveTextContent(/CL/i);
  })
  
  it('Should cancel editing with cancel button', async ()=>{
    const {container} = render(
      <MockedProvider addTypename={false} mocks={[mockData]}>
        <TxTable data={transactions} />
      </MockedProvider>
      )
      const editButton = screen.getByTestId('transaction-5d5c1f747e01cd704f18f864-editButton');
      userEvent.click(editButton);

      const userInput = screen.getByDisplayValue(/employee3/i);
      userEvent.clear(userInput);
      userEvent.type(userInput, "Trey");
      const cancelButton = screen.getByText(/cancel/i);
      const newUser = screen.getByDisplayValue(/trey/i);
      //checks to see if Trey has been typed in the document
      expect(newUser).toBeInTheDocument();

      userEvent.click(cancelButton);
      const updatedEditButton = await screen.findByTestId('transaction-5d5c1f747e01cd704f18f864-editButton');

      expect(newUser).not.toBeInTheDocument();
      expect(updatedEditButton).toBeInTheDocument();


  })
  it('Should delete transaction', async ()=>{
    const {container}=render(
      <MockedProvider mocks={[deleteMock]} addTypename={false}>
        <TxTable data={transactions} />
      </MockedProvider>
      )

      const button = screen.getByTestId('transaction-5d5c1f747e01cd704f18f863-deleteButton')

      userEvent.click(button)

      const deleteTransactionMutation = deleteMock.newData;
      await waitFor(() => expect(deleteTransactionMutation).toHaveBeenCalled());



  })

})
