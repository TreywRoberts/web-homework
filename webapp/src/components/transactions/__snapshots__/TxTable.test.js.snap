// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions Table Should cancel editing with cancel button 1`] = `
.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  padding: 1rem;
}

.emotion-0 table {
  border-collapse: collapse;
  width: 100%;
}

.emotion-0 th,
.emotion-0 td {
  border: 1px solid #ffffff;
  text-align: center;
  padding: 8px;
  font-size: 2rem;
}

.emotion-0 th {
  background-color: #f6f4e9;
}

.emotion-0 td {
  background-color: #f2f5fb;
}

.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 5px;
}

.emotion-0 form td:last-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.emotion-0 form * {
  font-size: 28px;
}

<div>
  <div
    class="app-container"
  >
    <form>
      <table
        class="emotion-0"
      >
        <tbody>
          <tr
            class="header"
          >
            <th>
              User ID
            </th>
            <th>
              Description
            </th>
            <th>
              Merchant ID
            </th>
            <th>
              Debit
            </th>
            <th>
              Credit
            </th>
            <th>
              Amount
            </th>
            <th>
              Actions
            </th>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f863"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-userId"
            >
              employee4
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-description"
            >
              cleaningsupplies
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-debit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-credit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-amount"
            >
              150
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f864"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-userId"
            >
              employee3
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-amount"
            >
              250
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f865"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-userId"
            >
              employee5
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-amount"
            >
              100
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</div>
`;

exports[`Transactions Table Should cancel editing with cancel button 2`] = `
.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  padding: 1rem;
}

.emotion-0 table {
  border-collapse: collapse;
  width: 100%;
}

.emotion-0 th,
.emotion-0 td {
  border: 1px solid #ffffff;
  text-align: center;
  padding: 8px;
  font-size: 2rem;
}

.emotion-0 th {
  background-color: #f6f4e9;
}

.emotion-0 td {
  background-color: #f2f5fb;
}

.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 5px;
}

.emotion-0 form td:last-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.emotion-0 form * {
  font-size: 28px;
}

<div>
  <div
    class="app-container"
  >
    <form>
      <table
        class="emotion-0"
      >
        <tbody>
          <tr
            class="header"
          >
            <th>
              User ID
            </th>
            <th>
              Description
            </th>
            <th>
              Merchant ID
            </th>
            <th>
              Debit
            </th>
            <th>
              Credit
            </th>
            <th>
              Amount
            </th>
            <th>
              Actions
            </th>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f863"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-userId"
            >
              employee4
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-description"
            >
              cleaningsupplies
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-debit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-credit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-amount"
            >
              150
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr>
            <td>
              <input
                id="user_id"
                placeholder="User"
                required=""
                type="text"
                value="Trey"
              />
            </td>
            <td>
              <select
                id="description"
                placeholder="Description"
                type="text"
              >
                <option
                  value="Food"
                >
                  Food
                </option>
                <option
                  value="Fun"
                >
                  Fun
                </option>
                <option
                  value="Gas"
                >
                  Gas
                </option>
              </select>
            </td>
            <td>
              <input
                id="merchant_id"
                placeholder="Merchant"
                required=""
                type="text"
                value="walmart"
              />
            </td>
            <td>
              <input
                id="debit"
                name="paymentType"
                placeholder="Debit"
                type="radio"
                value="false"
              />
            </td>
            <td>
              <input
                id="credit"
                name="paymentType"
                placeholder="Credit"
                type="radio"
                value="true"
              />
            </td>
            <td>
              <input
                id="amount"
                placeholder="Amount"
                required=""
                type="number"
                value="250"
              />
            </td>
            <td>
              <button
                type="submit"
              >
                Save
              </button>
              <button
                type="button"
              >
                Cancel
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f865"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-userId"
            >
              employee5
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-amount"
            >
              100
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</div>
`;

exports[`Transactions Table Should cancel editing with cancel button 3`] = `
.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  padding: 1rem;
}

.emotion-0 table {
  border-collapse: collapse;
  width: 100%;
}

.emotion-0 th,
.emotion-0 td {
  border: 1px solid #ffffff;
  text-align: center;
  padding: 8px;
  font-size: 2rem;
}

.emotion-0 th {
  background-color: #f6f4e9;
}

.emotion-0 td {
  background-color: #f2f5fb;
}

.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 5px;
}

.emotion-0 form td:last-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.emotion-0 form * {
  font-size: 28px;
}

<div>
  <div
    class="app-container"
  >
    <form>
      <table
        class="emotion-0"
      >
        <tbody>
          <tr
            class="header"
          >
            <th>
              User ID
            </th>
            <th>
              Description
            </th>
            <th>
              Merchant ID
            </th>
            <th>
              Debit
            </th>
            <th>
              Credit
            </th>
            <th>
              Amount
            </th>
            <th>
              Actions
            </th>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f863"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-userId"
            >
              employee4
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-description"
            >
              cleaningsupplies
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-debit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-credit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-amount"
            >
              150
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f864"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-userId"
            >
              employee3
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-amount"
            >
              250
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f865"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-userId"
            >
              employee5
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-amount"
            >
              100
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</div>
`;

exports[`Transactions Table Should delete transaction 1`] = `
<button
  data-testid="transaction-5d5c1f747e01cd704f18f863-deleteButton"
>
  delete
</button>
`;

exports[`Transactions Table Should delete transaction 2`] = `
.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  padding: 1rem;
}

.emotion-0 table {
  border-collapse: collapse;
  width: 100%;
}

.emotion-0 th,
.emotion-0 td {
  border: 1px solid #ffffff;
  text-align: center;
  padding: 8px;
  font-size: 2rem;
}

.emotion-0 th {
  background-color: #f6f4e9;
}

.emotion-0 td {
  background-color: #f2f5fb;
}

.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 5px;
}

.emotion-0 form td:last-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.emotion-0 form * {
  font-size: 28px;
}

<div>
  <div
    class="app-container"
  >
    <form>
      <table
        class="emotion-0"
      >
        <tbody>
          <tr
            class="header"
          >
            <th>
              User ID
            </th>
            <th>
              Description
            </th>
            <th>
              Merchant ID
            </th>
            <th>
              Debit
            </th>
            <th>
              Credit
            </th>
            <th>
              Amount
            </th>
            <th>
              Actions
            </th>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f863"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-userId"
            >
              employee4
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-description"
            >
              cleaningsupplies
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-debit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-credit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-amount"
            >
              150
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f864"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-userId"
            >
              employee3
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-amount"
            >
              250
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f865"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-userId"
            >
              employee5
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-amount"
            >
              100
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</div>
`;

exports[`Transactions Table Should delete transaction 3`] = `
<button
  data-testid="transaction-5d5c1f747e01cd704f18f863-deleteButton"
>
  delete
</button>
`;

exports[`Transactions Table Should delete transaction 4`] = `
.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  padding: 1rem;
}

.emotion-0 table {
  border-collapse: collapse;
  width: 100%;
}

.emotion-0 th,
.emotion-0 td {
  border: 1px solid #ffffff;
  text-align: center;
  padding: 8px;
  font-size: 2rem;
}

.emotion-0 th {
  background-color: #f6f4e9;
}

.emotion-0 td {
  background-color: #f2f5fb;
}

.emotion-0 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 5px;
}

.emotion-0 form td:last-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.emotion-0 form * {
  font-size: 28px;
}

<div>
  <div
    class="app-container"
  >
    <form>
      <table
        class="emotion-0"
      >
        <tbody>
          <tr
            class="header"
          >
            <th>
              User ID
            </th>
            <th>
              Description
            </th>
            <th>
              Merchant ID
            </th>
            <th>
              Debit
            </th>
            <th>
              Credit
            </th>
            <th>
              Amount
            </th>
            <th>
              Actions
            </th>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f863"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-userId"
            >
              employee4
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-description"
            >
              cleaningsupplies
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-debit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-credit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f863-amount"
            >
              150
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f863-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f864"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-userId"
            >
              employee3
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f864-amount"
            >
              250
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f864-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
          <tr
            data-testid="transaction-5d5c1f747e01cd704f18f865"
          >
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-userId"
            >
              employee5
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-description"
            >
              refund
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-merchant"
            >
              walmart
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-debit"
            >
              false
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-credit"
            >
              true
            </td>
            <td
              data-testid="transaction-5d5c1f747e01cd704f18f865-amount"
            >
              100
            </td>
            <td>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-editButton"
                type="button"
              >
                edit
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-deleteButton"
              >
                delete
              </button>
              <button
                data-testid="transaction-5d5c1f747e01cd704f18f865-numeralButton"
              >
                To Numeral
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </form>
  </div>
</div>
`;
